packages/compartment-mapper/src/parse-cjs-shared-export-wrapper.js

```diff
@@ -0,0 +1,111 @@
+// @ts-check
+const { apply } = Reflect;
+const { freeze, keys, create, hasOwnProperty } = Object;
+
+/**
+ * @param {Object} object
+ * @param {string} key
+ * @returns {boolean}
+ */
+const has = (object, key) => apply(hasOwnProperty, object, [key]);
+
+/**
+ * ModuleEnvironmentRecord wrapper
+ * Creates shared export processing primitives to be used both Location and Archive usecases of cjs
+ *
+ * @param {Object} moduleEnvironmentRecord
+ * @param {Compartment} compartment
+ * @param {Record<string, string>} resolvedImports
+ * @returns {{
+ *   module: { exports: any },
+ *   moduleExports: any,
+ *   afterExecute: Function,
+ *   require: Function,
+ * }}
+ */
+export const wrap = (moduleEnvironmentRecord, compartment, resolvedImports) => {
+  // This initial default value makes things like exports.hasOwnProperty() work in cjs.
+  moduleEnvironmentRecord.default = create(
+    compartment.globalThis.Object.prototype,
+  );
+
+  // Set all exported properties on the defult and on the namespace simultaneously for import *
+  // Root namespace is only accessible for imports. Requiring from cjs gets the default field of the namespace
+  const assignProp = (prop, value) => {
+    // TODO Consider __esModule handling
```

Comment:
There's a separate issue for this. Also, I don't see what we should do differently to handle it at this time.

Reply:
It is possible that `__esModule` was only helpful for ESM on CJS, not CJS on ESM. It would be good to rule that possibility out definitively.

Reply:
I'll start working on that later this week and I'll keep digging until I find an example that differs between using __esModule and not.

Reply:
It might be easier to page into memory what problem `__esModule` was created to exist and check whether that problem exists on ESM. I could be convinced that we can ignore it with a very short proof.